### PR TITLE
Make things a lot more strict

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by kevmoo** [comment]
@srawlins – I'd love your thoughts about how I did this. A parser writer I am not...

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+## 2.1.0
+
+* Improve strict spec compliance of `>` handling by always encoding as `&gt;`
+  – unless preceded by `/`.
+* Improve strict spec compliance for `blockquote` but always putting the closing
```
Comment (srawlins) [inline CHANGELOG.md]: but -> by?

File: lib/src/html_renderer.dart [hunk 2]
```diff
@@ -55,13 +72,23 @@ class HtmlRenderer implements NodeVisitor {
   }
 
   void visitText(Text text) {
-    buffer.write(text.text);
+    var content = text.text;
+    if (const ['p', 'li'].contains(_lastVisitedTag)) {
```
Comment (srawlins) [inline lib/src/html_renderer.dart]: What a weird rule, can you link to the commonmark spec behind this?

File: lib/src/html_renderer.dart [hunk 3]
```diff
@@ -91,7 +120,24 @@ class HtmlRenderer implements NodeVisitor {
   }
 
   void visitElementAfter(Element element) {
+    if (const ['blockquote', 'ul', 'ol'].contains(element.tag)) {
```
Comment (srawlins) [inline lib/src/html_renderer.dart]: Haha again with the weird rule. Commonmark citation?

File: lib/src/inline_parser.dart [hunk 4]
```diff
@@ -40,8 +40,10 @@ class InlineParser {
     TextSyntax(r'&[#a-zA-Z0-9]*;'),
     // Encode "&".
     TextSyntax(r'&', sub: '&amp;'),
-    // Encode "<". (Why not encode ">" too? Gruber is toying with us.)
```
Comment (srawlins) [inline lib/src/inline_parser.dart]: Bob's original comments in this repo are classic. Alas.

File: tool/stats.dart [hunk 5]
```diff
@@ -84,6 +84,10 @@ String _unitOutput(Iterable<DataCase> cases) => cases.map((dataCase) => '''
 ${dataCase.input}<<<
 ${dataCase.expectedOutput}''').join();
 
+/// Set this to `true` and rerun `--update-files` to ease finding easy strict
+/// fixes.
+const _improveStrict = false;
```
Comment (srawlins) [inline tool/stats.dart]: Ha. I like this.

File: lib/src/html_renderer.dart (hunk 3 above)
Comment (kevmoo) [reply lib/src/html_renderer.dart]: It's all about blocks within blocks. Updated the logic. Much better now.

File: lib/src/html_renderer.dart (hunk 2 above)
Comment (kevmoo) [reply lib/src/html_renderer.dart]: It's all about blocks within blocks. Updated the logic. Much better now.

File: CHANGELOG.md (hunk 1 above)
Comment (kevmoo) [reply CHANGELOG.md]: fixed

**Pull request merged**